### DISCUSSION TITLE
Problems with Tasmota and Tasmotizer

### DISCUSSION THREAD

**Discussion by ssgnoe** [arendst/Tasmota#10483] [Q&A] [answered]
Hey there,
I'm flashing Gosund SP 111 with Tasmota and Tasmotizer. If I'm using Tasmota 6.3.0, everything is working fine (I did find out, because I have older SP111 flashed with this version some time ago and they are working since then). If I'm using a newer version of Tasmota, the ESP does not start. If I flash the newer version OTA on the old working devices, they crash and do not start again. No response on Button clicks etc. I tied 7.2.0, 8.x.x and 9.x.x. Where is my fault? I'd like to use the actual version of Tasmota in best case in German version, but no chance. Tried Tiny Version, Englisch, German...
Thanks for your help!

**Comment by meingraham** [comment, marked as answer]
Did you follow the upgrade migration version path? You cannot upgrade directly from 6.x to 9.x. You have to go through intermediate versions due to changes to settings structures, GPIO configuration, etc.

https://tasmota.github.io/docs/Upgrading/#upgrade-flow

**Reply by ssgnoe** [reply]
Ok, I think there was the problem by upgrading. I think I did not correct. But by flashing it new, I think I do not need the upgrading Guide?
This is the bigger problem I have.
Does the Upgrade Flow meen, I have to upgrade on version after another? So from 6.3.0 to 6.7.1, then to 7.2.0...?

Thanks!

**Comment by meingraham** [comment]
Save your config as JSON output before you start using `decode-config`.

Upgrade from 6.3 to 6.7.1, then to 7.2, and then... per the flow. You can web OTA upgrade pointing OTAURL to each version's link provided in the Tasmota docs migration article. Otherwise, you can download each version (including minimal) and perform a local file OTA (loading the intermediate minimal) for each version. And, for safe measure, `Reset 6` in between each version.

You can restore settings manually and reference the JSON file as you go. Note that some `SetOption` were retired and later re-used for a different setting, so take care to check that any that you used still are relevant and used for the same thing.

You can restore from the JSON once you are on the latest Tasmota version but consider what I mentioned above. I have found that when restoring from an older version JSON, if the load fails, it provides the setting that is no longer compatible. I then edit the JSON to remove that setting and try again. There may be two or three that you have to try and edit since there have been many changes since 6.3.

When loading settings from a file, `decode-config` only changes the settings you have defined within the JSON input - this can also be a subset, other settings will be kept as is. The tool always retrieves the binary configuration file (.dmp) from device (or existing file), patches the settings based in the input file, and writes back the changed binary. This file then gets restored to the device, one can use a subset of settings (i.e., your specific configuration) as input (JSON) without impacting the existing device settings (e.g., Wi-Fi or MQTT).

**Reply by ssgnoe** [reply]
Thank you, I think the settings are no problem, because the Plugs only know my MQTT and Wifi Settings, so no problem when they are lost. For now I know how to upgrade working ones, but how to flash a new one with the latest version(s)? Why this does not work?

**Comment by meingraham** [comment]
Be sure that you always erase flash prior to flashing via the serial interface.

If you flash the new device with 6.3, does it function properly?

If you configure the device as "generic" with no GPIO assignments, does it at least boot and allow you to get to the web UI? If so, then the problem could be the GPIO configuration/template. For some of these manufacturers, even though the device says it's the same on the outside, sometimes the internals change on newer revisions. If it doesn't boot, then you may not have a properly flashed device (can happen) and try again from scratch.

If not of this resolves your issue, you may have a bad unit.

**Reply by ssgnoe** [reply]
If I flash 6.3 the devices work as expected, so I get a sonoff-xxx Wifi, can connect and push my Wifi settings. Device is connecting to my Wifi and everything is fine.
Flashing same way with Tasmotizer a newer version (without any settings), the device does not boot (I think, because no LED turns on and normaly it blinks). Relay not working on button push is a normal problem before I can configure as Blitzwolf Device.
I tried many times to flash, also tried OTA without Tasmotizer (some weeks ago) and had same behaviour.
I think on newer versions it also should start in AP Mode right?

**Comment by meingraham** [comment]
> start in AP mode...

Not necessarily. I don't know what the default module is for the Tasmota binaries (used to me the Sonoff Basic). It's been a while since I've flashed a device using the serial interface. Anyway, the "default" GPIO config could conflict with the operation... but I don't think that's what's going on in your case.

I wonder if you start with 6.3 and then step through the migration version flow using Tasmotizer (with erase flash selected) for each version... at which version it no longer functions. That might give a clue as to where the problem may lay.

**Reply by ssgnoe** [reply]
I flashed newly on 6.3 working, I also tried every last version before 7.x 8.x and 9.x and none of them worked...

**Comment by ssgnoe** [comment]
I had the time to check the Upgrade Flow and yeah! I was able to upgrade from 6.xx to 9.2.0. So it's working now. I also flashed to Novostella Devices yesterday via tuya_convert and then upgraded, worked finde. Don't know why I could not flash the newest directly after soldering, but if tuya convert ist working I do not have to soldier anymore. Thank you all for the help, I'm very happy now!
